src/graph/GoExecutor.cpp

```diff
@@ -1010,100 +1103,68 @@ bool GoExecutor::processFinalResult(RpcResponse &rpcResp, Callback cb) const {
                                    std::make_shared<ResultSchemaProvider>(schema.second));
                            });
         }
-
-        if (tagSchema.empty() && edgeSchema.empty()) {
-            continue;
-        }
-
         for (auto &vdata : resp.vertices) {
             DCHECK(vdata.__isset.edge_data);
             auto tagData = vdata.get_tag_data();
+            auto srcId = vdata.get_vertex_id();
             for (auto &edata : vdata.edge_data) {
-                auto it = edgeSchema.find(edata.type);
-                DCHECK(it != edgeSchema.end());
-                RowSetReader rsReader(it->second, edata.data);
-                auto iter = rsReader.begin();
                 auto edgeType = edata.type;
-                while (iter) {
-                    std::vector<SupportedType> colTypes;
-                    bool saveTypeFlag = false;
-                    auto &getters = expCtx_->getters();
-                    getters.getAliasProp = [&iter,
-                                            &spaceId,
-                                            &edgeType,
-                                            &saveTypeFlag,
-                                            &colTypes,
-                                            &edgeSchema,
-                                            srcid = vdata.get_vertex_id(),
-                                            this](const std::string &edgeName,
-                                                  const std::string &prop) -> OptVariantType {
-                        EdgeType type;
-                        auto found = expCtx_->getEdgeType(edgeName, type);
-                        if (!found) {
-                            return Status::Error(
-                                    "Get edge type for `%s' failed in getters.", edgeName.c_str());
+                for (auto& edge : edata.edges) {
+                    auto dstId = edge.get_dst();
+                    // Check if duplicate
+                    if (distinct_) {
+                        auto ret = uniqResult->emplace(std::make_pair(dstId, edge.get_props()));
```

Comment:
It seems not correct here, if we do calculations with the results, the final result could be duplicated.

Reply:
It should be the same with current logic. 
Could you give me a case for it?

Reply:
I got your point.  You are right,  we should do distinct operation on final result.